providers/base/bin/cpuid.py

```diff
@@ -139,6 +140,26 @@
         "Whisky Lake":      ['0x806eb', '0x806ec'],
         }
 
+# Server chips use different generations, so skip them
+INTEL_GEN_TO_NUM = {
+        "Raptorlake":       13,
+        "Alderlake":        12,
+        "Rocket Lake":      11,
+        "Tiger Lake":       11,
+        "Comet Lake":       10,
+        "Amber Lake":       10,
+        "Ice Lake":         10,
+        "Whiskey Lake":     10,
+        "Canon Lake":       8,
+        "Coffee Lake":      8,
+        "Kaby Lake":        7,
+        "Skylake":          6,
```

Comment:
If this is truly NOT meant for Server CPUs, you'll need to also break out the ID look up to identify "Skylake" vs "Skylake Xeon", look up the IDs and break them out accordingly.  

And then what about things like Coffee Lake, Kaby Lake, Rocket Lake, etc Xeon-E,D,W chips?

Reply:
Ya, this is why I planned to not address server chips. I don't know the requirements of people testing Xeon products, and I figure they'd be able to piggyback on my changes if the need exists.

Reply:
Yeah, I guess the question is where does this get run?  So the way this is, if you run this against a skylake, for example, on a skylake server (Xeon Scalable 1st Gen), that will then incorrectly report your generation of 6 since the skylake entry has codes for both:
"Skylake":          ['0x406e3', '0x506e3', '0x50654', '0x50652'],
** IIRC the first two are Skylake Y and Skylake DT/H/S (https://en.wikichip.org/wiki/intel/microarchitectures/skylake_(client)#CPUID)

also, Some of those ONLY list server CPUs since I wrote it for server, expecting that if anyone needed any additional client CPUs, they'd add those themselves... and I didn't differentiate Client vs Server per platform, as I said...

I guess I'm just trying to imagine the cleanest way to do this, because for now, matching any server cpu platform on your lookup table will return the client cpu generation incorrectly.

So Maybe it's a good idea to also break the CPUID table down as part of this?